lib/wallaby/httpclient.ex

```diff
@@ -23,18 +23,22 @@ defmodule Wallaby.HTTPClient do
   def request(method, url, params, [{:encode_json, false} | _]) do
     make_request(method, url, params)
   end
+  def request(method, url, params, [{:max_try, max} | _]) do
+    make_request(method, url, params, {0, max})
+  end
   def request(method, url, params, _opts) do
     make_request(method, url, Poison.encode!(params))
   end
 
-  defp make_request(method, url, body, retry_count\\0)
-  defp make_request(_, _, _, 5), do: raise "Wallaby had an internal issue with HTTPoison"
-  defp make_request(method, url, body, retry_count) do
+  defp make_request(method, url, body, _ \\ {0, 5})
+  defp make_request(_, _, _, {m, m}), do: raise "Wallaby had an internal issue with HTTPoison; all #{m} tries failed."
```

Comment:
`m` doesn't really speak, how I understand it this is `max_tries` or `tries` ?

Reply:
If `tries` equals to `max_tries`, it means all tries have been exhausted but still no success.

Reply:
I know what it means :) I'm saying that `m` as a variable name doesn't really communicate that because `m` doesn't say `tries` or `max_tries` but then again, I'm not sure we need this code at all (see bigger review comment)

Reply:
I do not get your point; it is a stop condition. Actually I not much familiar with Elixir, but I find this code works fine.

Reply:
the point is not about the code working fine or not, but about readability. Good code is readable and easy to understand sot hat in maintenance, bug fixes and new features it's easier to do